app/code/Bulbulatory/Recommendations/etc/db_schema.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
+<schema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Setup/Declaration/Schema/etc/schema.xsd">
+    <table name="bulbulatory_recommendations" resource="default" engine="innodb" comment="Bulbulatory Recommendations">
+        <column xsi:type="int" name="id" padding="10" unsigned="true" nullable="false" identity="true" comment="Affiliate link ID"/>
+        <column xsi:type="text" name="reference" comment="Customer Reference"/>
+        <column xsi:type="varchar" name="email" length="128" unsigned="false" nullable="false" comment="Email address, where the link was sent"/>
+        <column xsi:type="varchar" name="hash" length="255" unsigned="false" nullable="false" comment="Hash will be used in the affiliate links"/>
+        <column xsi:type="smallint" name="state" padding="1" default="1" unsigned="false" nullable="false" comment="1 - confirmed / 0 - unconfirmed"/>
+        <column xsi:type="timestamp" name="created_at" nullable="false" comment="Hash will be used in the affiliate links"/>
+        <column xsi:type="datetime" name="confirmation_at" unsigned="false" nullable="true" comment="Hash will be used in the affiliate links"/>
+        <constraint xsi:type="primary" referenceId="PRIMARY">
+            <column name="id"/>
+        </constraint>
+        <constraint xsi:type="unique" referenceId="UNIQUE">

```

Comment:
`referenceId` is a unique reference to a constraint so what if in the future there will be a new field in the table that should also be unique? You can't use the `UNIQUE` reference second time, `UNIQUE2` or something like that is not very readable. It's a good practice to add at least a column name to it and maybe also a table name so by looking at the constraint you know where it applies. If it is eg. `BULBULATORY_RECOMMENDATIONS_HASH_UNIQUE` you know straight away that it applies to column `hash` in the table `bulbulatory_recommendations`.

Reply:
@TM18 

> referenceId is a unique reference to a constraint so what if in the future there will be a new field in the table that should also be unique? You can't use the UNIQUE reference second time, UNIQUE2 or something like that is not very readable. It's a good practice to add at least a column name to it and maybe also a table name so by looking at the constraint you know where it applies. If it is eg. BULBULATORY_RECOMMENDATIONS_HASH_UNIQUE you know straight away that it applies to column hash in the table bulbulatory_recommendations.

I misunderstood the documentation. I changed _referenceId_ value as you wrote.